tests/unittest/grpc/test_grpc_optional.py

```
@@ -132,3 +135,31 @@ def test_smg_server_startup_failure_cleans_up(monkeypatch, failure_point):
+    """Absent bindings surface as ImportError, which is what serve.py catches.
+
+    ``trtllm-serve --grpc --grpc_protocol openengine`` wraps the import of
+    ``tensorrt_llm.grpc.openengine.server`` in ``except ImportError`` to
+    re-raise it as a ClickException carrying the
+    ``pip install "tensorrt_llm[openengine]"`` hint. If the module ever fails
+    with something outside that hierarchy, the hint silently stops reaching
+    users, so pin the contract the handler depends on.
+    """
+    real_import = builtins.__import__
+
+    def import_without_openengine(name, globals=None, locals=None, fromlist=(), level=0):
```

Ruff (0.16.3)

[error] Function argument `globals` is shadowing a Python builtin

(A002)